letsencrypt-win-simple/Plugin/SanByDomainPlugin.cs

```diff
@@ -0,0 +1,288 @@
+        }
+
+        public override void PrintMenu()
+        {
+            Console.WriteLine(" D: Generate a single San certificate by Domain with subdomains.");
+        }
+
+        public override void HandleMenuResponse(string response, List<Target> targets)
+        {
+            if (response == "d")
+            {
+                Console.WriteLine("Running SanByDomain Plugin");
+                Log.Information("Running SanByDomain Plugin");
+
+                List<Domain> domains = new List<Domain>();
+
+                using (var iisManager = new ServerManager())
+                {
+                    foreach (var site in iisManager.Sites)
+                    {
+                        foreach (var binding in site.Bindings)
+                        {
+                            if (!String.IsNullOrEmpty(binding.Host))
+                            {
+                                string domain = GetDomain(binding.Host);
+
+                                //Hosts can appear more than once with different protocols. We only need it once
+                                var dom = domains.FirstOrDefault(o => o.Host == binding.Host);
+                                if (dom != null)
+                                {
+                                    if (!dom.HasHttpsBinding)
+                                        dom.HasHttpsBinding = (binding.Protocol == "https");
+                                }
+                                else
+                                {
+                                    domains.Add(new Domain
+                                    {
+                                        DomainName = domain,
+                                        Host = binding.Host.Trim(),
+                                        IsDomain = (domain == binding.Host.ToLowerInvariant().Trim()),
+                                        HasHttpsBinding = (binding.Protocol == "https"),
+                                        WebRootPath = site.Applications["/"].VirtualDirectories["/"].PhysicalPath,
+                                        SiteId = site.Id
+                                    });
+                                }
+                            }
+                        }
+                    }
+                }
+
+                var i = 0;
+                var curDom = "";
+                foreach (var item in domains.OrderBy(o => o.DomainName).ThenBy(o => o.IsDomain == false).ThenBy(o => o.Host))
+                {
+                    if (item.DomainName != curDom)
+                    {
+                        i++;
+                        var dom = domains.FirstOrDefault(o => o.Host == item.Host);
+                        dom.DomainId = i;
+                        Console.WriteLine();
+                    }
+
+                    var line = (((item.DomainId > 0) ? "  " + item.DomainId + ": " : "  -- ") + item.Host + ((item.HasHttpsBinding) ? " (https)" : "")).PadRight(50) + " - " + item.DomainName;
```

Comment:
You have multiple if's in there. The ? : syntax is not used in the rest of the app so it shouldn't be used here.